### PR TITLE
Package details page warning for SemVer is inconsistent with other places

### LINKED ISSUE #4552: Package details page warning for SemVer is inconsistent with other places.
Expected:
image
This is what we see while uploading a SemVer package

Actual:
image

### PR DESCRIPTION
**Pull request by xavierdecoster**
Fixes #4552

### CODE REVIEW

File: src/NuGetGallery/ViewModels/PackageViewModel.cs [hunk 1]
```diff
@@ -24,7 +25,13 @@ public PackageViewModel(Package package)
             Version = String.IsNullOrEmpty(package.NormalizedVersion) ?
                 NuGetVersionFormatter.Normalize(package.Version) :
                 package.NormalizedVersion;
-            
+
+            HasSemVer2Version = NuGetVersion.Parse(_fullVersion).IsSemVer2;
+            HasSemVer2Dependency = package.Dependencies.ToList()
```
Comment (loic-sharma) [inline src/NuGetGallery/ViewModels/PackageViewModel.cs]: Just to double-check - does `Dependencies` include dependencies of dependencies?

Comment (xavierdecoster) [reply src/NuGetGallery/ViewModels/PackageViewModel.cs]: Those are the top-level dependencies only.

Comment (loic-sharma) [reply src/NuGetGallery/ViewModels/PackageViewModel.cs]: Oh. Does that mean that a package is not considered a semver 2 package if one of its dependencies has a semver 2 dependency?

Comment (xavierdecoster) [reply src/NuGetGallery/ViewModels/PackageViewModel.cs]: Yep! See the non-goals in the SemVer2 spec for nuget.org ;) https://github.com/NuGet/Home/wiki/SemVer2-support-for-nuget.org-%28server-side%29#non-goals

Comment (loic-sharma) [reply src/NuGetGallery/ViewModels/PackageViewModel.cs]: I see, thanks for the info! :)

**Pull request merged**